### PR TITLE
Rav 4755 nvl synapse support

### PR DESCRIPTION
**Pull request by saranshmanglani**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java [hunk 1]
```diff
@@ -2799,6 +2799,21 @@ public static Object ifNull(Object first, Object second) {
...
 
+  /***
+   * If first operand is not null isNull will return first operand
+   * else it will return second operand.
+   * @param first operand
+   * @param second operand
+   * @return Object
+   */
+  public static Object isNull(Object first, Object second) {
+    if (first != null) {
```
Comment (vjain061) [inline core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java:2810]: Use ternary operator

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibrary.java [hunk 2]
```diff
@@ -60,7 +60,9 @@
   /** A collection of operators that are in Teradata but not in standard SQL. */
   TERADATA("t"),
   /** A collection of operators that are in Snowflake but not in standard SQL. */
-  SNOWFLAKE("sf");
+  SNOWFLAKE("sf"),
+  /** A collection of operators that are in Snowflake but not in standard SQL. */
```
Comment (vjain061) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibrary.java:64]: Comment is same as Snowflake

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 3]
```diff
@@ -75,6 +79,12 @@ public MssqlSqlDialect(Context context) {
         }
         unparseFloor(writer, call);
         break;
+      case NVL:
```
Comment (vjain061) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:82]: Should we do it in raven side?

**Comment by vaibhavjain-dm** [comment]
All checks have failed please have a look.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -4399,17 +4399,6 @@ public void testTimestampFunctionRelToSql() {
         .ok(expectedSpark);
   }
 
-  @Test public void testTimestampPlusIntervalMonthFunctionWithArthOps() {
-    String query = "select \"hire_date\" + -10 * INTERVAL '1' MONTH from \"employee\"";
-    final String expectedBigQuery = "SELECT CAST(DATETIME_ADD(CAST(hire_date AS DATETIME), "
-        + "INTERVAL "
```
Comment (soma-mondal) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:4405]: Please revert this

**Review by soma-mondal — APPROVED** [review]

**Review by deepakDatametica — APPROVED** [review]

**Pull request merged**